examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py

```diff
@@ -49,6 +41,13 @@ class AirflowPollingSensorCursor:
 
 def build_airflow_polling_sensor(
     airflow_instance: AirflowInstance,
+    # Maps each dag to a dictionary of its tasks, and each task to the set of asset keys it produces.
+    keys_per_task_handle: Mapping[str, Mapping[str, Set[AssetKey]]],
+    key_per_dag_id: Mapping[str, AssetKey],
+    # Maps each dag to a dictionary of its tasks, and each task to a boolean indicating whether it has been migrated.
+    migration_state_per_task_handle: Mapping[str, Mapping[str, Optional[bool]]],
+    asset_keys_topological_order: Sequence[AssetKey],
+    check_keys_per_asset_key: Mapping[AssetKey, Set[AssetCheckKey]],
```

Comment:
there is a lot of "parameter clumping" in this PR, indicating some missing abstractions here. Seems like we need an intermediate object that contains all the interesting information we have accessed from the Airflow that is fast and easy-to-use, and then some elegant way of encoding information from an arbitrary asset graph subset

Reply:
yea we should just make the underlying serializable object readily available to get rid of this all I think

Reply:
Only two of these fields feel broadly applicable to any asset graph subset, as it were. I would still want to compute them alongside everything else in a single pass, but are you also imagining having an AssetGraphData thing which includes the more arbitrary data?